absl/strings/numbers.h

```diff
@@ -100,26 +100,80 @@ static const int kSixDigitsToBufferSize = 16;
 // Required buffer size is `kSixDigitsToBufferSize`.
 size_t SixDigitsToBuffer(double d, char* buffer);
 
+// These functions are here for simplicity and static type dispatching to FastU?Int(32|64)ToBuffer
+template<typename int_type>
+char* FastIntToBufferBase(int_type i, char* buffer) = delete;
+template<> inline char* FastIntToBufferBase<int32_t>(int32_t i, char* buffer)
+{
+  return FastInt32ToBuffer(i, buffer);
+}
+template<> inline char* FastIntToBufferBase<uint32_t>(uint32_t i, char* buffer)
+{
+  return FastUInt32ToBuffer(i, buffer);
+}
+template<> inline char* FastIntToBufferBase<int64_t>(int64_t i, char* buffer)
+{
+  return FastInt64ToBuffer(i, buffer);
+}
+template<> inline char* FastIntToBufferBase<uint64_t>(uint64_t i, char* buffer)
+{
+  return FastUInt64ToBuffer(i, buffer);
+}
+
+// These functions are here for simplicity and static type dispatching to safe_strtou?(32|64)_base
+template<typename int_type>
+bool safe_strto_base(absl::string_view text, int_type* value, int base) = delete;
+template<> inline bool safe_strto_base<int32_t>(absl::string_view text, int32_t* value, int base)
+{
+  return safe_strto32_base(text, value, base);
+}
+template<> inline bool safe_strto_base<uint32_t>(absl::string_view text, uint32_t* value, int base)
+{
+  return safe_strtou32_base(text, value, base);
+}
+template<> inline bool safe_strto_base<int64_t>(absl::string_view text, int64_t* value, int base)
+{
+  return safe_strto64_base(text, value, base);
+}
+template<> inline bool safe_strto_base<uint64_t>(absl::string_view text, uint64_t* value, int base)
+{
+  return safe_strtou64_base(text, value, base);
+}
+
+template<typename any_type>
+struct Identity final
```

Comment:
This already exists in absl/base/internal/identity.h - maybe it should just move to meta/type_traits? Regardless, for now you should use that one.

Reply:
Fixed